pkg/skaffold/deploy/helm.go

```diff
@@ -162,14 +162,43 @@ func (h *HelmDeployer) Dependencies() ([]string, error) {
 			continue
 		}
 
-		chartDepsDir := filepath.Join(release.ChartPath, "charts")
+		chartDepsDirs := []string{
+			"charts",
+			"tmpcharts",
+		}
+
+		lockFiles := []string{
+			"Chart.lock",
+			"requirements.lock",
+		}
 
-		// We can always add a dependency if it is not contained in our chartDepsDir.
+		// We can always add a dependency if it is not contained in our chartDepsDirs.
 		// However, if the file is in our chartDepsDir, we can only include the file
 		// if we are not running the helm dep build phase, as that modifies files inside
 		// the chartDepsDir and results in an infinite build loop.
+		// We additionally exclude ChartFile.lock (Helm 3) and requirements.lock (Helm 2)
+		// since they also get modified on helm dep build phase
 		isDep := func(path string, info walk.Dirent) (bool, error) {
-			return !info.IsDir() && (!strings.HasPrefix(path, chartDepsDir) || r.SkipBuildDependencies), nil
+			if info.IsDir() {
+				return false, nil
+			}
+			if r.SkipBuildDependencies {
+				return true, nil
+			}
+
+			for _, v := range chartDepsDirs {
+				if strings.HasPrefix(path, filepath.Join(release.ChartPath, v)) {
+					return false, nil
+				}
+			}
+
+			for _, v := range lockFiles {
+				if strings.EqualFold(info.Name(), v) {
```

Comment:
Would this miss any other requirements.lock say in a python app? 
Shd we look for `reuirements.lock` in the chartPath?

Reply:
Can we also a test case for this?

Reply:
are you saying that I might miss some `.lock` file or that I'll erroneously include it?
I'm currently removing the `.lock` files by name without depending on the path and AFAIK `Chart.lock` and `requirements.lock` are colocated with the respective `.yaml` files, which is always in `chartPath

Reply:
discussed offline